src/core/include/units/magnitude.h

```diff
@@ -225,7 +325,23 @@ static constexpr bool is_base_power_pack_valid = all_base_powers_valid<BPs...> &
  */
 template<BasePower auto... BPs>
   requires (detail::is_base_power_pack_valid<BPs...>)
-struct magnitude {};
+struct magnitude {
+  // Whether this magnitude represents an integer.
+  static constexpr bool is_magnitude_integral = (detail::is_integral(BPs) && ...);
+  friend constexpr bool is_integral(magnitude) { return is_magnitude_integral; }
```

Comment:
Do we need two sets of those? Isn't a hidden friend function enough? Also, it looks a bit strange to apply the same naming convention to member variables and functions.